src/griffe/mixins.py

```diff
@@ -131,3 +133,45 @@ def is_implicitely_exported(self) -> bool:
             True or False.
         """
         return self.parent.exports is None  # type: ignore[attr-defined]
+
+
+class SerializationMixin:
+    """A mixin that adds de/serialization conveniences."""
+
+    def as_json(self, full: bool = False, indent: int = 2, **kwargs: Any) -> str:
+        """Return this object's data as a JSON string.
+
+        Parameters:
+            full: Whether to return full info, or just base info.
+            indent: Indentation level for JSON output.
+            **kwargs: Additional serialization options passed to encoder.
+
+        Returns:
+            A string.
+        """
+        from griffe.encoders import JSONEncoder  # avoid circular import
+
+        return json.dumps(self, cls=JSONEncoder, indent=indent, full=full, **kwargs)
+
+    @classmethod
+    def from_json(cls: Type[_ObjType], json_string: str, **kwargs) -> _ObjType:
+        """Create an instance of this class from a JSON string.
+
+        Parameters:
+            json_string: JSON to decode into Object.
+            **kwargs: Additional options passed to decoder.
+
+        Returns:
+            An Object instance.
+
+        Raises:
+            TypeError: When the json_string does not represent and object
+                       of the class from which this classmethod has been called.
```

Comment:
Doesn't the Google parser complain about the indentation here :sweat_smile:? It should be indented once more, not aligned with the beginning of the sentence.

Reply:
lol.  yeah, I did that trying to satisfy the parser :joy: ... will tone it back a bit :)